sql/rpl_master.cc

```diff
@@ -1284,49 +1284,54 @@ bool reset_master(THD *thd, bool unlock_global_read_lock) {
 
   @param thd      Pointer to THD object for the client thread executing the
                   statement.
-  @param file     Name of the current bin log.
-  @param pos      Position int he current bin log.
-  @param gtid_executed        Logged gtids in binlogs.
-  @param gtid_executed_length Length of gtid_executed string.
-  @param snapshot_hlc  HLC when the snapshot was taken
+  @param ss_info  Snapshot context that contains binlog file/pos,
+                  executed gtids and snapshot id
   @param need_ok  [out] Whether caller needs to call my_ok vs it having been
                   done in this function via my_eof.
 
   @retval false success
   @retval true failure
 */
-bool show_master_offset(THD *thd, const char *file, ulonglong pos,
-                        const char *gtid_executed, int gtid_executed_length,
-                        ulonglong snapshot_hlc, bool *need_ok) {
+bool show_master_offset(THD *thd, snapshot_info_st &ss_info,
+                        bool *need_ok) {
   Protocol *protocol = thd->get_protocol();
   DBUG_ENTER("show_master_offset");
   List<Item> field_list;
   field_list.push_back(new Item_empty_string("File", FN_REFLEN));
   field_list.push_back(
       new Item_return_int("Position", 20, MYSQL_TYPE_LONGLONG));
   field_list.push_back(
-      new Item_empty_string("Gtid_executed", gtid_executed_length));
+      new Item_empty_string("Gtid_executed", ss_info.gtid_executed.length()));
 
-  if (snapshot_hlc != 0) {
+  if (ss_info.snapshot_hlc != 0) {
     field_list.push_back(
         new Item_return_int("Snapshot_HLC", 20, MYSQL_TYPE_LONGLONG));
   }
 
+  if (ss_info.snapshot_id != 0) {
```

Comment:
In 5.6, the snapshot_id came before snapshot_hlc. We should preserve the order here.

Reply:
Done